test/integration/entry-integration.js

```diff
@@ -480,48 +476,87 @@ describe('Entry Api', () => {
+      let xSpaceEnabledContentType
+
+      before(async () => {
+        // The default space has x-space enabled in it through the feature flags
+        xSpaceEnabledSpace = await getDefaultSpace()
+        xSpaceEnabledEnvironment = await createTestEnvironment(
+          xSpaceEnabledSpace,
+          'Test Cross Space'
+        )
+        await waitForEnvironmentToBeReady(xSpaceEnabledSpace, xSpaceEnabledEnvironment)
+        xSpaceEnabledContentType = await xSpaceEnabledEnvironment.getContentType(
+          TestDefaults.contentType.withCrossSpaceReferenceId
+        )
+      })
 
-      // Creating a new space that doesn't have the xspace feature enabled
-      xSpaceDisabledSpace = await createTestSpace(initClient(), 'XSpaceDisabledEntry')
-      xSpaceDisabledEnvironment = await createTestEnvironment(
-        xSpaceDisabledSpace,
-        'Testing Environment'
-      )
-      xSpaceDisabledContentType = await xSpaceDisabledEnvironment.createContentTypeWithId(
-        generateRandomId('test-content-type'),
-        contentTypeData
-      )
-      await xSpaceDisabledContentType.publish()
+      after(async () => {
+        if (xSpaceEnabledEnvironment) {
+          await xSpaceEnabledEnvironment.delete()
+        }
+      })
+
+      test('can create, publish, unpublish and delete an entry with xspace references', async () => {
+        return xSpaceEnabledEnvironment
+          .createEntry(xSpaceEnabledContentType.sys.id, entryData)
+          .then((entry) => {
+            expect(entry.isDraft(), 'entry is in draft').ok
+            expect(entry.fields.title['en-US']).equals('this is the title', 'original title')
+            return entry.publish().then((publishedEntry) => {
+              expect(publishedEntry.isPublished(), 'entry is published').ok
+              return publishedEntry.unpublish().then((unpublishedEntry) => {
+                expect(unpublishedEntry.isDraft(), 'entry is back in draft').ok
+                return publishedEntry.delete()
+              })
+            })
+          })
+      })
     })
 
-    after(async () => {
-      // Cleaning up the xSpaceDisabled Space
-      await xSpaceDisabledEnvironment.deleteEntry(xSpaceDisabledEntryId)
-      await xSpaceDisabledContentType
-        .unpublish()
-        .then((unpublishedContentType) => unpublishedContentType.delete())
+    describe('When cross-space references feature is disabled for the space', () => {
+      let xSpaceDisabledSpace
+      let xSpaceDisabledEnvironment
+      let xSpaceDisabledContentType
+      let xSpaceDisabledEntryId
 
-      await xSpaceDisabledSpace.delete()
+      before(async () => {
+        // Creating a new space that has the x-space feature disabled
+        xSpaceDisabledSpace = await createTestSpace(initClient(), 'XSpaceDisabledEntry')
+        xSpaceDisabledEnvironment = await xSpaceDisabledSpace.getEnvironment('master')
+        await waitForEnvironmentToBeReady(xSpaceDisabledSpace, xSpaceDisabledEnvironment)
```

Comment:
`await createTestSpace` should return a fully created space with a ready `master` environment. 
I would expect that you don't necessarily have to wait for the `master` environment to b ready.